subsys/mgmt/mcumgr/lib/cmd/img_mgmt/include/img_mgmt/img_mgmt.h

```diff
@@ -64,9 +64,9 @@ extern struct img_mgmt_state g_img_mgmt_state;
 
 /** Represents an individual upload request. */
 struct img_mgmt_upload_req {
-	unsigned long long image;	/* 0 by default */
-	unsigned long long off;		/* -1 if unspecified */
-	unsigned long long size;	/* -1 if unspecified */
+	uint32_t image;	/* 0 by default */
+	uint32_t off;	/* UINT32_MAX if unspecified */
+	uint32_t size;	/* UINT32_MAX if unspecified */
```

Comment:
This seems fine, but would it be expected that for other systems e.g. a 64-bit platforms, there might be a very large image transferred, or an image transferred with a very large offset?

Reply:
> This seems fine, but would it be expected that for other systems e.g. a 64-bit platforms, there might be a very large image transferred, or an image transferred with a very large offset?

Perfectly it would `size_t`, I guess that could be done, just need to do some #ifdefs to switch zcbor encoding on the fields.

> doesn't these offsets are within the image?

Yes they are. The data here is provided by mcumgr that only knows image, does not know flash layout.